core/src/HIP/Kokkos_HIP_Space.cpp

```diff
@@ -195,6 +211,65 @@ void* HIPHostPinnedSpace::impl_allocate(
 
   return ptr;
 }
+
+void* HIPManagedSpace::allocate(const size_t arg_alloc_size) const {
+  return allocate("[unlabeled]", arg_alloc_size);
+}
+void* HIPManagedSpace::allocate(const char* arg_label,
+                                const size_t arg_alloc_size,
+                                const size_t arg_logical_size) const {
+  return impl_allocate(arg_label, arg_alloc_size, arg_logical_size);
+}
+void* HIPManagedSpace::impl_allocate(
+    const char* arg_label, const size_t arg_alloc_size,
+    const size_t arg_logical_size,
+    const Kokkos::Tools::SpaceHandle arg_handle) const {
+  void* ptr = nullptr;
+
+  if (arg_alloc_size > 0) {
+    if (is_first_hip_managed_allocation.exchange(false) &&
+        Kokkos::show_warnings()) {
+      if (!Kokkos::Impl::hip_driver_check_page_migration(m_device)) {
+        std::cerr
+            << R"warning(The combination of device and system configuration does not support page migration between device and host. HIPManagedSpace might not work as expected. Please refer to the Rocm documentation on unified/managed memory\n)warning"
```

Comment:
I would prefer if we would obey some line length limits here and in other places.